### DISCUSSION TITLE
extending App?

### DISCUSSION THREAD

**Discussion by kud** [vercel/next.js#17799] [Help] [answered]
Hello,

I would like to understand what's the point of extending `App` in `_app.js` file?

I see some tutorials saying we should code this:

```
import React from 'react'
import App from 'next/app'

class MyApp extends App {}
```

but the official tutorial says this:

```
function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}
```

I understand that we've got both class and function here, but I still don't understand the point of extending `App`?

Thank you 😊

**Comment by leerob** [comment, marked as answer]
Next.js uses the App component to initialize pages. You can override it and control the page initialization. Which allows you to do amazing things like:

- Persisting layout between page changes
- Keeping state when navigating pages
- Custom error handling using componentDidCatch
- Inject additional data into pages
- Add global CSS

https://nextjs.org/docs/advanced-features/custom-app

**Reply by kud** [reply]
Hello @leerob and thank you for your quick answer!

I was probably not so clear but my question was about extending `next/app` (or not), not about what the point of customising `_app.js`.

**Reply by leerob** [reply]
That is just the difference between a class and a function. 

> The extends keyword is used in class declarations or class expressions to create a class that is a child of another class.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/extends

**Reply by leerob** [reply]
In general, you should follow the examples in the documentation 👍

**Reply by kud** [reply]
Thank you!
